lib/ui/window/platform_configuration.h

```diff
@@ -0,0 +1,126 @@
+// Copyright 2013 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+#ifndef FLUTTER_LIB_UI_WINDOW_PLATFORM_CONFIGURATION_H_
+#define FLUTTER_LIB_UI_WINDOW_PLATFORM_CONFIGURATION_H_
+
+#include <memory>
+#include <set>
+#include <string>
+#include <unordered_map>
+#include <vector>
+
+#include "flutter/fml/time/time_point.h"
+#include "flutter/lib/ui/semantics/semantics_update.h"
+#include "flutter/lib/ui/window/platform_message.h"
+#include "flutter/lib/ui/window/pointer_data_packet.h"
+#include "flutter/lib/ui/window/viewport_metrics.h"
+#include "flutter/lib/ui/window/window.h"
+#include "third_party/tonic/dart_persistent_value.h"
+
+namespace tonic {
+class DartLibraryNatives;
+
+// So tonic::ToDart<std::vector<int64_t>> returns List<int> instead of
+// List<dynamic>.
+template <>
+struct DartListFactory<int64_t> {
+  static Dart_Handle NewList(intptr_t length) {
+    return Dart_NewListOf(Dart_CoreType_Int, length);
+  }
+};
+
+}  // namespace tonic
+
+namespace flutter {
+class FontCollection;
+class PlatformMessage;
+class Scene;
+
+// Must match the AccessibilityFeatureFlag enum in framework.
+enum class AccessibilityFeatureFlag : int32_t {
+  kAccessibleNavigation = 1 << 0,
+  kInvertColors = 1 << 1,
+  kDisableAnimations = 1 << 2,
+  kBoldText = 1 << 3,
+  kReduceMotion = 1 << 4,
+  kHighContrast = 1 << 5,
+};
+
+class PlatformConfigurationClient {
+ public:
+  virtual std::string DefaultRouteName() = 0;
+  virtual void ScheduleFrame() = 0;
+  virtual void Render(Scene* scene) = 0;
+  virtual void UpdateSemantics(SemanticsUpdate* update) = 0;
+  virtual void HandlePlatformMessage(fml::RefPtr<PlatformMessage> message) = 0;
+  virtual FontCollection& GetFontCollection() = 0;
+  virtual void UpdateIsolateDescription(const std::string isolate_name,
+                                        int64_t isolate_port) = 0;
+  virtual void SetNeedsReportTimings(bool value) = 0;
+  virtual std::shared_ptr<const fml::Mapping> GetPersistentIsolateData() = 0;
+  virtual std::unique_ptr<std::vector<std::string>>
+  ComputePlatformResolvedLocale(
+      const std::vector<std::string>& supported_locale_data) = 0;
+
```

Comment:
This class should probably be preventing copies (deleting the relevant constructors).

Reply:
Done.